### PR TITLE
i18n(zh-tw): update `why-astro.mdx` translation

### PR DESCRIPTION
**Pull request by mingjunlu**
#### Description

- Updates Traditional Chinese translation of `src/content/docs/zh-tw/concepts/why-astro.mdx`

#### Related issues & labels

- ~~Closes #~~
- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/zh-tw/concepts/why-astro.mdx [hunk 1]
```diff
@@ -1,45 +1,53 @@
...
+- **[伺服器優先](/zh-tw/basics/rendering-modes/)**：避免在客戶端執行耗費資源的渲染。
+- **[預設零 JS](/zh-tw/basics/astro-components/)**：減少拖慢網站速度的客戶端 JavaScript。
+- **[內容合集](/zh-tw/guides/content-collections/)**：統一管理、驗證 Markdown 內容，同時提供 TypeScript 型別安全。
+- **[客製化](/zh-tw/guides/integrations-guide/)**：Tailwind、MDX 等數以百計的整合任君挑選。
 
-## 以內容為主
+## 設計理念
 
-**Astro 是專為豐富內容的網站設計的。** 這包括大部分的行銷網站、出版網站、文檔網站、部落格、作品集以及部分電子商務網站。
+為何要在眾多網頁框架中選擇 Astro 呢？以下五個核心設計理念，有助於解釋為什麼我們要打造 Astro、它解決了甚麼問題以及為什麼 Astro 可能是專案或團隊的最佳選擇。
```
Comment (liruifengv) [inline src/content/docs/zh-tw/concepts/why-astro.mdx:24]: Missed `Astro is…` here?

File: src/content/docs/zh-tw/concepts/why-astro.mdx [hunk 2]
```diff
@@ -50,30 +58,31 @@ SPA 模型有它的優勢，然而代價是額外的複雜度及折衷的性能
...
-Astro 被設計成比其他 UI 框架及語言還簡單。一個重要原因是 Astro 是設計用來在伺服器而非瀏覽器上渲染。這就代表你無須擔心像是：hooks（React）、stale closures（還是 React）、refs（Vue）、observables（Svelte）、atoms、selectors、reactions 或 derivations。由於伺服器並沒有反應性，所以這些複雜性就不見了。
+Astro 設計成比其他 UI 框架及語言還簡單，其中一個重要原因是：Astro 是用來在伺服器而非瀏覽器上渲染。這就代表你毋需擔心像：hook（React）、stale closure（還是 React）、ref（Vue）、observable（Svelte）、atom、selector、reaction 或 derivation。由於使用者無法在伺服器端渲染時與頁面互動，所以這些複雜度都消失了。
 
-我們最喜歡的說法之一是：**選擇性地加入複雜度。** 我們在設計 Astro 時盡可能的在開發者體驗中去除「必須的複雜度」，特別是當你第一次使用這套框架。你可以在 Astro 中只用 HTML 和 CSS 建立一個「Hello World」範例網站。當你需要建立更強大的的東西時，你可以在使用過程中逐漸增加新功能及 API。
+我們最喜歡的其中一句話是：**選擇加入複雜度（opt in to complexity）**。設計 Astro 時，我們盡可能地在開發者體驗中去除「必須的複雜度」，特別是當你第一次使用這套框架時。你可以在 Astro 中只用 HTML 和 CSS 建立「Hello World」範例網站，而當你需要建立更強大的的東西時，隨時可以在過程中逐漸增加新功能及 API。
 
+### 注重開發者
 
-## 靈活且齊全
+我們深信，當開發者享受使用 Astro 時，它才算是一個成功的專案。Asstro 提供一切所需，幫助你建立網站。
```
Comment (liruifengv) [inline src/content/docs/zh-tw/concepts/why-astro.mdx:82]:
```suggestion
我們深信，當開發者享受使用 Astro 時，它才算是一個成功的專案。Astro 提供一切所需，幫助你建立網站。
```

File: src/content/docs/zh-tw/concepts/why-astro.mdx (hunk 1 above)
Comment (mingjunlu) [reply src/content/docs/zh-tw/concepts/why-astro.mdx:24]: Thanks! Added here: https://github.com/withastro/docs/pull/7361/commits/2bef8099d74bb315493d67aca032bdc7056e54b3#diff-411f02195e31915769d02061af65a5c9167741c7dc87cd6d5d2fe0283ba84c5dR26

**Review by liruifengv — APPROVED** [review]
LGTM. Thanks!

**Review by dreyfus92 — APPROVED** [review]
Thank you peeps! LGTM ✅

**Pull request merged**